tests/test_weaver_snapshot.py

```
@@ -0,0 +1,1727 @@
+    lock.write_text("a previous run left this", encoding="utf-8")
+
+    with weaver_snapshot._lock_file(lock):
+        pass
+
+    assert lock.read_text(encoding="utf-8") == "a previous run left this", (
+        "the lock file was truncated; the lock is the flock, not the contents"
+    )
+
+
+def test_a_failed_publication_leaves_the_destination_as_it_was(
+    tmp_path: Path, monkeypatch: pytest.MonkeyPatch
```

Ruff (0.16.2)

[warning] Unused function argument: `monkeypatch`

(ARG001)